bigquery/google/cloud/bigquery/job.py

```diff
@@ -1218,11 +1218,53 @@ def from_api_repr(cls, resource, client):
         return job
 
 
-class _AsyncQueryConfiguration(object):
-    """User-settable configuration options for asynchronous query jobs.
+class QueryJobConfig(object):
+    """Configuration options for query jobs.
 
-    Values which are ``None`` -> server defaults.
+    All properties in this class are optional. Values which are ``None`` ->
+    server defaults.
     """
+
+    def __init__(self):
+        self._properties = {}
+        # TODO(swast): Default to standard SQL syntax.
+
+    def to_api_repr(self):
+        """Build an API representation of the copy job config.
+
+        :rtype: dict
+        :returns: A dictionary in the format used by the BigQuery API.
+        """
+        return copy.deepcopy(self._properties)
+
+    @classmethod
+    def from_api_repr(cls, resource):
+        """Factory: construct a job configuration given its API representation
+
+        :type resource: dict
+        :param resource:
+            An extract job configuration in the same representation as is
+            returned from the API.
+
+        :rtype: :class:`google.cloud.bigquery.job.ExtractJobConfig`
+        :returns: Configuration parsed from ``resource``.
+        """
+        config = cls()
+        config._properties = copy.deepcopy(resource)
+        return config
+
+    use_legacy_sql = _TypedApiResourceProperty(
+        'use_legacy_sql', 'useLegacySql', bool)
+    """See
+    https://cloud.google.com/bigquery/docs/\
+    reference/v2/jobs#configuration.query.useLegacySql
+    """
+
+    dry_run = _TypedApiResourceProperty('dry_run', 'dryRun', bool)
+    """See
+    https://g.co/cloud/bigquery/docs/reference/v2/jobs#configuration.dryRun
+    """
+
```

Comment:
Shouldn't all the fields below become `ApiResourceProperty`s?

Reply:
Yes, but I only did the minimum for replacing `run_sync_query`. I'll do the rest when I replace `run_async_query`. I started doing both, but that was becoming way too big a change with all the tests that needed updates.